apps/plea/models.py

```diff
@@ -639,3 +651,332 @@ class DataValidation(models.Model):
+        null=True, blank=True)
+
+    event_type = models.CharField(
+        choices=EVENT_TYPE_CHOICES,
+        verbose_name="event type",
+        help_text="Identified the area of the application that the event happened in",
+        max_length=255)
+
+    event_subtype = models.CharField(
+        choices=EVENT_SUBTYPE_CHOICES,
+        verbose_name="reason for failure",
+        help_text="The specific reason for the event",
+        max_length=255)
+
+    event_trace = models.CharField(
+        max_length=4000,
+        verbose_name="error detail",
+        help_text="This detail about the reason for this event may be useful for developers to debug import issue",
+        blank=True, null=True)
+
+    event_data = HStoreField(
+        verbose_name="Event data",
+        help_text="If there was a failure and data fields were found, they are stored here to debug",
+        null=True, blank=True)
+
+    extra_data_hash = models.CharField(
+        verbose_name="extra data hash",
+        help_text="If the event caused a change to the extra_data then store the hash of it for debugging",
+        max_length=32, default="")
+
+    event_datetime = models.DateTimeField(
+        verbose_name="event date and time",
+        help_text="The time at which the event occurred",
+        auto_now_add=True)
+
+    @property
+    def urn(self):
+        """URN may come from the case or the extra data. Used in the admin."""
+        extra_urn = case_urn = None
+
+        if hasattr(self, "extra_data"):
+            if "urn" in self.extra_data:
+                extra_urn = self.extra_data["urn"]
+
+        if hasattr(self, "case"):
+            if hasattr(self.case, "urn"):
+                case_urn = self.case.urn
+
+        if case_urn != extra_urn:
+            return "CONFLICTED"
+        else:
+            return case_urn or extra_urn
+
+    @property
+    def initiation_type(self):
+        """May come from the case or the event data. Used in the admin"""
+        itype_edata = itype_attr = None
+
+        if hasattr(self, "extra_data"):
+            if "initiation_type" in self.extra_data:
+                itype_edata = self.extra_data["initiation_type"]
+
+        if hasattr(self, "case"):
+            if hasattr(self.case, "initiation_type"):
+                itype_attr = [
+                    i[1]
+                    for i in INITIATION_TYPE_CHOICES
+                    if self.case.initiation_type == i[0]
+                ][0]
+
+        if itype_attr != itype_edata:
+            return "CONFLICTED"
+        else:
+            return itype_attr or itype_edata
+
+    @classmethod
+    def populate(cls, *args, **kwargs):
+        """
+        Use of _meta.get_all_field_names() needs changing for Django 1.9
+        http://stackoverflow.com/questions/2170228/iterate-over-model-instance-field-names-and-values-in-template
+
+        TODO: Visitor pattern made sense when I started this work, not as much anymore
+        """
```

Comment:
Also, it's listed as a classmethod here, but it always seems to be called on an instance. eg. https://github.com/ministryofjustice/manchester_traffic_offences_pleas/pull/324/files?diff=unified#diff-f898d2457745cf2610fc8d01dc9bbd4eR379